cocos/base/CCEventDispatcher.cpp

```diff
@@ -222,6 +222,83 @@ EventDispatcher::~EventDispatcher()
     removeAllEventListeners();
 }
 
+
+void EventDispatcher::visitTree(Node* node)
```

Comment:
Could you explain new algorithm and provide profile results. What is the speedup.

Reply:
```
for (; i < childrenCount; i++)
{
	child = children.at(i);
	if (child && child->getLocalZOrder() < 0)
	{
		if (child->getUserData() == 0) // here!!!
			continue;
		visitTree(child);
	}
	else
		break;
}
```

I think its easy idea, if the tree had N nodes, and the sceneGraphListeners had M listeners, you could visit from M listener node to root, and color the path.
second, visit from root to child by the colored path. it could be avoid to visit useless path.

in my test proj, I had tree which contain 6K nodes, but listeners only about 600. I thought the case which listnener much less than nodes was in general.  O(N) -> O(c*M), c is the coefficient about the avg children and node in path. but in the worst c*M = N

Reply:
sorry, I could not provide the test case, it contain too many complex widgets.
but I test in win32, the old method cost 8ms, and cost 3ms when use the new one.

Reply:
```
		if ((nodeData & 1) == 1)
			_globalZOrderNodeMap[node->getGlobalZOrder()].push_back(node);
```
```
if (_nodeListenersMap.find(node) != _nodeListenersMap.end())
        {
            _globalZOrderNodeMap[node->getGlobalZOrder()].push_back(node);
        }
```

and stl unordered_map also more expensive than bit mask.
when the N nodes is big, the cost need be consider